packages/@aws-cdk/aws-ec2/lib/machine-image.ts

```diff
@@ -402,13 +402,16 @@ export class AmazonLinuxImage extends GenericSSMParameterImage {
     let virtualization: AmazonLinuxVirt | undefined;
     let storage: AmazonLinuxStorage | undefined;
 
-    if (generation === AmazonLinuxGeneration.AMAZON_LINUX_2022) {
-      kernel = AmazonLinuxKernel.KERNEL5_X;
+    if (generation === AmazonLinuxGeneration.AMAZON_LINUX_2022 || generation === AmazonLinuxGeneration.AMAZON_LINUX_2023) {
+      kernel = (props && props.kernel) || AmazonLinuxKernel.DEFAULT;
```

Comment:
So this seems like a breaking change. For some current users their kernel version goes from 5_X to DEFAULT. Is that intentional and what are the implications of that? What are the implications generally of the SSM parameter name changing? Does it matter?

Additionally I'm not sure if it was intentional that when props.kernel was defined, and AMAZON_LINUX_2022 was the generation, then the kernel version got set to 5_X.

Reply:
See this comment for how we want to handle this

https://github.com/aws/aws-cdk/issues/21011#issuecomment-1372170948